beacon_node/network/src/sync/range_sync/chain.rs

```diff
@@ -1096,27 +999,23 @@ impl<T: BeaconChainTypes> SyncingChain<T> {
             }
 
             if let Entry::Vacant(entry) = self.batches.entry(epoch) {
-                if let Some(peer) = idle_peers.pop() {
-                    let batch_type = network.batch_type(epoch);
-                    let optimistic_batch = BatchInfo::new(&epoch, EPOCHS_PER_BATCH, batch_type);
-                    entry.insert(optimistic_batch);
-                    self.send_batch(network, epoch, peer)?;
-                }
+                let batch_type = network.batch_type(epoch);
+                let optimistic_batch = BatchInfo::new(&epoch, EPOCHS_PER_BATCH, batch_type);
+                entry.insert(optimistic_batch);
+                self.send_batch(network, epoch)?;
             }
             return Ok(KeepChain);
         }
 
-        while let Some(peer) = idle_peers.pop() {
+        loop {
```

Comment:
this loop mirrors the logic you have above but missing this comment:
```
// Note: for this function to not infinite loop we must:
// - If `include_next_batch` returns Some we MUST increase the count of batches that are
//   accounted in the `BACKFILL_BATCH_BUFFER_SIZE` limit in the `matches!` statement of
//   that function.
```
it does seem like `send_batch()` is limited by `BATCH_BUFFER_SIZE` in this case. I think the `while let Some(batch_id)` version is cleaner.

Reply:
Good find! The duplicate code between forward and backfill sync is a bit annoying